packages/sdk/src/mls/utils/mlsutils.ts

```diff
@@ -1,12 +1,40 @@
 import { check } from '@river-build/dlog'
 import { IStreamStateView } from '../../streamStateView'
 
+import { StreamTimelineEvent } from '../../types'
+
 export type ExtractMlsExternalGroupResult = {
     externalGroupSnapshot: Uint8Array
     groupInfoMessage: Uint8Array
     commits: { commit: Uint8Array; groupInfoMessage: Uint8Array }[]
 }
 
+function commitFromEvent(
+    event: StreamTimelineEvent,
+): { commit: Uint8Array; groupInfoMessage: Uint8Array } | undefined {
+    const payload = event.remoteEvent?.event.payload
+    if (payload?.case !== 'memberPayload') {
+        return undefined
+    }
+    if (payload?.value?.content.case !== 'mls') {
+        return undefined
+    }
+
+    const mlsPayload = payload.value.content.value
+    switch (mlsPayload.content.case) {
+        case 'externalJoin':
```

Comment:
Not sure if this is a valid suggestion, because I haven't tried it yet, but
- if you check if group info and commit are present first, then
- does TypeScript narrow down `content.case` to all cases that have both of them present, so that
- you can switch on it, and then
- check if you handled all the cases with `logNever`?